### PR TITLE
Tests: Update isAmazonLinux2() logic

### PR DESCRIPTION
**Pull request by bkhouri**
The ProcessInfo extension to determine if a host is running on Amazon Linux 2 was slightly flawed as it did not take into account the double-quote (") surrounding the values.

Update the logic to look at "PRETTY_NAME" since the ID can be used in multiple Amazon Linux OS'es, and use a regular expression in the logic.

### CODE REVIEW

**Comment by bkhouri** [comment]
@swift-ci test

**Comment by bkhouri** [comment]
I pushed a change that update the tests to add coverage with difference scenarios.

@swift-ci test

**Comment by bkhouri** [comment]
@swift-ci test windows

File: Sources/_InternalTestSupport/ProcessInfo+hostutils.swift [hunk 1]
```diff
@@ -22,14 +22,16 @@ extension ProcessInfo {
                 return false
             }
         }
-        let lines = contentString.components(separatedBy: .newlines)
-        for line in lines {
-            if line.starts(with: "ID=") {
-                let id = line.replacingOccurrences(of: "ID=", with: "").trimmingCharacters(in: .whitespacesAndNewlines)
-                if id == "amzn" { // ID for Amazon Linux is "amzn"
+        do {
+            let nameRegex = try Regex("PRETTY_NAME=\"?Amazon Linux 2.*\"?")
```
Comment (jakepetroules) [inline Sources/_InternalTestSupport/ProcessInfo+hostutils.swift]: I suggest you just do an exact match without the regex -- as is, this will incorrectly match Amazon Linux 2023 (due to the greedy match), which is the next version of Amazon Linux that the Swift project will later switch to, and notably does _not_ have the posix_spawn issues that cause us to special case AL2 in the first place.

**Review by jakepetroules — CHANGES_REQUESTED** [review]

File: Sources/_InternalTestSupport/ProcessInfo+hostutils.swift (hunk 1 above)
Comment (MaxDesiatov) [reply Sources/_InternalTestSupport/ProcessInfo+hostutils.swift]: I'd rather have this merged now and edge cases dealt with later in separate PRs. This PR is blocking toolchain builds and AL2 failures is one of the reasons we don't have published toolchains since 22nd of June.

File: Tests/BasicsTests/ProcessInfoTests.swift [hunk 2]
```diff
@@ -50,10 +50,73 @@ struct ProcessInfoExtensionTests {
...
+                        ID="amzn"
+                        ID_LIKE="centos rhel fedora"
+                        VERSION_ID="2"
+                        PRETTY_NAME=Amazon Linux 2
+                        ANSI_COLOR="0;33"
+                        CPE_NAME="cpe:2.3:o:amazon:amazon_linux:2"
+                        HOME_URL="https://amazonlinux.com/"
+                        SUPPORT_END="2026-06-30"
+                        """,
+                    expected: true
```
Comment (jakepetroules) [inline Tests/BasicsTests/ProcessInfoTests.swift]: Here is AL2023:

```
NAME="Amazon Linux"
VERSION="2023"
ID="amzn"
ID_LIKE="fedora"
VERSION_ID="2023"
PLATFORM_ID="platform:al2023"
PRETTY_NAME="Amazon Linux 2023.6.20250107"
ANSI_COLOR="0;33"
CPE_NAME="cpe:2.3:o:amazon:amazon_linux:2023"
HOME_URL="https://aws.amazon.com/linux/amazon-linux-2023/"
DOCUMENTATION_URL="https://docs.aws.amazon.com/linux/"
SUPPORT_URL="https://aws.amazon.com/premiumsupport/"
BUG_REPORT_URL="https://github.com/amazonlinux/amazon-linux-2023"
VENDOR_NAME="AWS"
VENDOR_URL="https://aws.amazon.com/"
SUPPORT_END="2028-03-15"
```

**Review by MaxDesiatov — CHANGES_REQUESTED** [review]
the cross-repo job is still failing: https://ci.swift.org/job/swift-PR-toolchain-amazonlinux2/79/console
```
18:46:28  ✘ Test run with 499 tests in 91 suites failed after 1168.385 seconds with 99 issues (including 88 known issues).
```

File: Tests/BasicsTests/ProcessInfoTests.swift (hunk 2 above)
Comment (bkhouri) [reply Tests/BasicsTests/ProcessInfoTests.swift]: Thanks for this.

**Comment by bkhouri** [comment]
@swift-ci test

**Comment by bkhouri** [comment]
@swift-ci test

**Comment by bkhouri** [comment]
I ran cross-repository testing
- https://github.com/swiftlang/swift/pull/65101#issuecomment-3069736307
- https://ci.swift.org/job/swift-PR-toolchain-amazonlinux2/81/

**Comment by bkhouri** [comment]
@swift-ci test windows

**Review by jakepetroules — APPROVED** [review]

**Comment by bkhouri** [comment]
I ran cross-repository testing

- https://github.com/swiftlang/swift/pull/65101#issuecomment-3071612379
- https://ci.swift.org/job/swift-PR-toolchain-amazonlinux2/82/

@swift-ci test

**Comment by bkhouri** [comment]
@swift-ci test windows

**Comment by bkhouri** [comment]
@swift-ci test self hosted windows

**Comment by MaxDesiatov** [comment]
@swift-ci test self hosted windows

**Comment by MaxDesiatov** [comment]
@swift-ci test self hosted windows

**Comment by MaxDesiatov** [comment]
@swift-ci test windows self hosted

**Comment by MaxDesiatov** [comment]
@swift-ci test self hosted windows

**Comment by bnbarham** [comment]
Merging to unblock toolchain builds
